backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/po/POTrlRepository.java

```diff
@@ -255,11 +287,71 @@ else if (po.isDynAttributeTrue(DYNATTR_TrlUpdateMode_UpdateIdenticalTrls))
 
 		final int updatedCount = DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
 		logger.debug("Updated {} translation records for {}", updatedCount, po);
+		final CacheMgt cacheMgt = CacheMgt.get();
+		cacheMgt.reset(toTrlTableName(tableName), po.get_ID());
 
-		//
 		return updatedCount >= 0;
 	}
 
+	public void updateBaseTableOnTranslationChange(@NonNull final PO po)
+	{
+		final POTrlInfo trlInfo = po.getPOInfo().getTrlInfo();
+		if (!trlInfo.isTranslations())
+		{
+			return;
+		}
+
+		final String trlTableName = po.get_TableName();
+		final String baseTableName = toBaseTableName(trlTableName);
+		if(DB.isDBColumnPresent(baseTableName, COLUMNNAME_AD_Element_ID))
+		{
+			return;
+		}
+
+		final String keyColumnName;
+		final String keyColumnValue;
+		final String languageColumnValue;
+		if(trlInfo.getKeyColumnNames().get(0).equalsIgnoreCase(I_AD_Language.COLUMNNAME_AD_Language))
+		{
+			keyColumnName = trlInfo.getKeyColumnNames().get(1);
+			keyColumnValue = String.valueOf(po.get_ID(1));
+			languageColumnValue = String.valueOf(po.get_ID(0));
+		}
+		else
+		{
+			keyColumnName = trlInfo.getKeyColumnNames().get(0);
+			keyColumnValue = String.valueOf(po.get_ID(0));
+			languageColumnValue = String.valueOf(po.get_ID(1));
+		}
+
+		final String baseLanguage = Language.getBaseLanguage().getAD_Language();
+		if(!Objects.equals(languageColumnValue, baseLanguage))
+		{
+			return;
+		}
+
+		final List<String> translatedColumnNames = getPOInfo(baseTableName).getTranslatedColumnNames();
+		final StringBuilder sqlSet = new StringBuilder();
+
+
+		for (final String columnName : translatedColumnNames)
+		{
+			final String sqlValue = convertValueToSql(po.get_Value(columnName));
+			sqlSet.append(columnName).append("=").append(sqlValue).append(", ");
+		}
+		sqlSet.deleteCharAt(sqlSet.lastIndexOf(","));
```

Comment:
that looks awkard.

when building the sqlSet, better do something like
if(sqlSet.length() > 0) sqlSet.append(", ");
sqlSet.append(columnName).append("=")...................